### DISCUSSION TITLE
Use default paramters for methods where it makes sense

### DISCUSSION THREAD

**Discussion by SeanFarrow** [rasmus/EventFlow#858] [Q&A]
It would be really nice if, where possible, EventFlow methods used default C# parameters.
I noticed this when looking at the way CancellationTokens are passed around.
Is this an area in which a PR would be accepted?
Thanks,
Sean.

**Comment by rasmus** [comment]
For the public APIs it would improve the API and thus a PR would be great.

Internally within EventFlow we really need to make sure that the cancellation tokens provided by developers are passed along, and using defaults there will make it harder to spot if we forget to pass one along.

**Comment by SeanFarrow** [comment]
Ok, how about the following as separate PR’s:
Firstly, add a default CancellationToken (i.e CancellationToken.None) to any public methods that take a CancellationToken.
My question here would be what do you consider as public? To my thinking, it would be any class implementing an interface, but feel free to correct my understanding.
Secondly, following on from this, ensure that any cancellationToken (including the default) is passed to private methods.
I’m happy to do the first this weekend and the second within the next few weeks.
Thanks,
Sean.
